src/CPU_Instruction_Set.md

```diff
@@ -1,151 +1,170 @@
- bit  n,r        | CB xx    |       8      | z01-  | test bit n
- bit  n,(HL)     | CB xx    |      12      | z01-  | test bit n
- set  n,r        | CB xx    |       8      | ----  | set bit n
- set  n,(HL)     | CB xx    |      16      | ----  | set bit n
- res  n,r        | CB xx    |       8      | ----  | reset bit n
- res  n,(HL)     | CB xx    |      16      | ----  | reset bit n
-
-## CPU Control instructions
-
-Mnemonic         | Encoding | Clock cycles | Flags | Description
------------------|----------|--------------|-------|-------------
- ccf             | 3F       |       4      | -00c  | cy=cy xor 1
- scf             | 37       |       4      | -001  | cy=1
- nop             | 00       |       4      | ----  | no operation
- halt            | 76       |     N*4      | ----  | halt until interrupt occurs (low power)
- stop            | 10 00    |       ?      | ----  | low power standby mode (VERY low power)
- di              | F3       |       4      | ----  | disable interrupts, IME=0
- ei              | FB       |       4      | ----  | enable interrupts, IME=1
-
-## Jump instructions
-
-Mnemonic         | Encoding | Clock cycles | Flags | Description
------------------|----------|--------------|-------|-------------
- jp   nn         | C3 nn nn |      16      | ----  | jump to nn, PC=nn
- jp   HL         | E9       |       4      | ----  | jump to HL, PC=HL
- jp   f,nn       | xx nn nn |    16/12     | ----  | conditional jump if nz,z,nc,c
- jr   PC+dd      | 18 dd    |      12      | ----  | relative jump to nn (PC=PC+8-bit signed)
- jr   f,PC+dd    | xx dd    |     12/8     | ----  | conditional relative jump if nz,z,nc,c
- call nn         | CD nn nn |      24      | ----  | call to nn, SP=SP-2, (SP)=PC, PC=nn
- call f,nn       | xx nn nn |    24/12     | ----  | conditional call if nz,z,nc,c
- ret             | C9       |      16      | ----  | return, PC=(SP), SP=SP+2
- ret  f          | xx       |     20/8     | ----  | conditional return if nz,z,nc,c
- reti            | D9       |      16      | ----  | return and enable interrupts (IME=1)
- rst  n          | xx       |      16      | ----  | call to 00,08,10,18,20,28,30,38
+TODO: links to optables and gbz80(7), and explain formatting
+
+Possible values for the bitfields:
+
+{{#bits 8 <
+	"r8"      0:"<code>b</code>" 1:"<code>c</code>" 2:"<code>d</code>" 3:"<code>e</code>" 4:"<code>h</code>" 5:"<code>l</code>" 6:"<code>[hl]</code>" 7:"<code>a</code>" ;
+	"r16"     0:"<code>bc</code>" 1:"<code>de</code>" 2:"<code>hl</code>" 3:"<code>sp</code>" ;
+	"r16stk"  0:"<code>bc</code>" 1:"<code>de</code>" 2:"<code>hl</code>" 3:"<code>af</code>" ;
+	"r16mem"  0:"<code>bc</code>" 1:"<code>de</code>" 2:"<code>hl+</code>" 3:"<code>hl-</code>" ;
+	"cond"    0:"<code>nz</code>" 1:"<code>z</code>" 2:"<code>nc</code>" 3:"<code>c</code>" ;
+	"b3"      0-7:"A 3-bit bit index" ;
+	"imm8"    0-7:"The following byte" ;
+	"imm16"   0-7:"The following two bytes, in little-endian order" ;
+}}
```

Comment:
nit: `vec3` isn't described here, but it's the operand type written out for `rst`. clearly rst doesn't take a `b3` or anything like that, so that makes sense. but it'd be helpful to have some words like `"vec3" 0-4: "The reset vector number. Multiply by 8 for the vector's address"` in this table.

Reply:
Yeah, that's indeed an accidental omission. I would suggest the following wording, though: `"rst's target address, divided by 8"`?